adapter/pkg/resource_types/dto.go

```diff
@@ -0,0 +1,147 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package resourcetypes
+
+// Subscription for struct subscription
+type Subscription struct {
+	SubscriptionID    int    `json:"subscriptionId"`
+	PolicyID          string `json:"policyId"`
+	APIID             int    `json:"apiId"`
+	AppID             int    `json:"appId" json:"applicationId"`
+	SubscriptionState string `json:"subscriptionState"`
+	TenantID          int
+	TenantDomain      string
+	TimeStamp         int64
```

Comment:
Shall we make these TenantID, TenantDomain, and TimeStamp fields optional, since those data are not present in the response received at the startup?

for ex: 
TenantID int `json:"tenanId,omitempty"`

Reply:
+1

Reply:
fixed by https://github.com/wso2/product-microgateway/pull/1559/commits/6b76d2d42a4d5b0cd4a1a88327f2e447bcf03883